packages/openapi-fetch/test/middleware/middleware.test.ts

```diff
@@ -443,3 +443,64 @@ test("type error occurs only when neither onRequest nor onResponse is specified"
   assertType<Middleware>({ onResponse });
   assertType<Middleware>({ onRequest, onResponse });
 });
+
+test("can return response directly from onRequest", async () => {
+  const customResponse = Response.json({});
+  const client = createObservedClient<paths>();
```

Comment:
```suggestion
  const client = createObservedClient<paths>({}, () => throw new Error("unexpected call to fetch"));
```

This way, we make sure we really do not call fetch.